tun_disabled.go

```diff
@@ -36,12 +53,69 @@ func (*disabledTun) DeviceName() string {
 }
 
 func (t *disabledTun) Read(b []byte) (int, error) {
-	<-t.block
-	return 0, io.EOF
+	r, ok := <-t.read
+	if !ok {
+		return 0, io.EOF
+	}
+
+	t.tx.Inc(1)
+	if l.Level >= logrus.DebugLevel {
+		t.logger.WithField("raw", prettyPacket(r)).Debugf("Write payload")
+	}
+
+	return copy(b, r), nil
```

Comment:
Just for some extra safety, consider:

```suggestion
	payloadLen = len(r)
	n := copy(b, r)
	if n != payloadLen {
		return payloadLen, fmt.Errorf("sent truncated payload %d/%d bytes", n, payloadLen)
	}
	
	return n, nil
```

Reply:
I'll add a check for this and also I'll check when we receive an ICMP Echo Request that it is <= mtu